app/controllers/api/admin/chemicals_controller.rb

```diff
@@ -0,0 +1,238 @@
+module Api
+  module Admin
+    class ChemicalsController < ApiController
+      before_action :authenticate_user!
+      before_action :set_up_errors
+      before_action :fetch_chemical, only: %i[show show_map update names delete_map create_map update_map]
+
+      # GET    /admin/chemicals
+      def index
+        respond_with Chemical.order_by(:id).all
```

Comment:
@mcumcu could you please add pagination?

Reply:
Pagination added to `/admin/chemicals`
commit: 4967b6eb